### ISSUE TITLE
Strange massive false positive case

### ISSUE THREAD

**Issue by zloirock** [eslint-community/eslint-plugin-es-x#244]
After updating `8.5.0` -> `8.6.0`, in the `core-js` repo I see

<img width="935" alt="Image" src="https://github.com/user-attachments/assets/ab772e8c-2e73-4690-9594-d109de4b4ec9" />

For example, on

```js
var array = [];
var constructor = array.constructor = {};
```

I see

```
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-every
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-filter
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-foreach
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-indexof
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-lastindexof
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-map
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-reduce
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-reduceright
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-some
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-copywithin
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-entries
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-fill
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-find
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-findindex
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-keys
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-values
  2:19  error  ES2016 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-includes
  2:19  error  ES2019 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-flat
  2:19  error  ES2022 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-at
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-findlast-findlastindex
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es/no-array-prototype-toreversed
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es/no-array-prototype-tosorted
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es/no-array-prototype-tospliced
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es/no-array-prototype-with
```

**Comment by ota-meshi** [comment]
Thank you for posting this issue! 
I fixed the bug. Could you please try using 8.6.1?
